### ISSUE TITLE
Automatic tests on pull requests

### ISSUE THREAD

**Issue by BenediktBurger** [PyMoDAQ/pymodaq_data#11]
For these packages without GUI (like data, utils etc.), tests are quite feasable, therefore I suggest, that the tests are run for pull requests, to ensure correct working for all changes.

**Comment by seb5g** [comment]
but they are already ??

**Comment by BenediktBurger** [comment]
In #10, no tests ran.

**Comment by BenediktBurger** [comment]
A quick check on the github [documentation](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#push):

- "push" works on pushed commits in this repository (so on your branches), but not on pushes to branches in other repositories (like mine)

What we use in pymeasure:
~~~
on:
  push:
    branches:
      - master
      - develop
  pull_request:
~~~

that way only pushes to the main or develop branch (no feature branches) are checked, or a pull request (maybe of a feature branch).

A quote from the documentation:
~~~
on:
  push:
    branches:
      - 'main'
      - 'releases/**'
~~~

that allows to test branches following a certain pattern (in your case "feature/", for example).

**Comment by seb5g** [comment]
nice! But then your issue is about PR while here you're talking about commits. Are we clear you 'd like tests to run on every push even on other branches? But then which branch should we include? Can we not do it all the time no matter the branch or fork?

**Comment by BenediktBurger** [comment]
Sorry for the confusion by talking about the different possibilities.

Right now, it is run on all pushes in **this repository** (so no forks).
The "push" section considers only this repository (no forks).

I suggest to run it on pull_requests (from this repo or other repos).

Additionally, it might be good to run it push to the master branch (so even code which enters outside a pull request, e.g. direct push) gets tested. This test will be run after pull requests, making sure, that the merged code works as expected.

Whether pushes to other branches of your repo trigger a test run or not is not that relevant.
If these branches are part of a pull request, the tests will be executed twice (once for the PR and once for the push), if I remember correctly.

Therefore, the Pymeasure tests are only for the master branch (and an unused "develop" branch).

**TL;DR**: Please add tests for pull requests. You might restrict tests for pushes at the same time, if you wish so.

**Comment by seb5g** [comment]
PR #13 is fixing this issue

**Comment by BenediktBurger** [comment]
Thanks.
